tests/unit-tests/scene/test_surface_stack.cpp

```diff
@@ -531,9 +531,7 @@ TEST_F(SurfaceStack, scene_observer_can_query_scene_within_surface_exists_notifi
     MockSceneObserver observer;
 
     auto const scene_query = [&]{
-        stack.for_each([&](std::shared_ptr<mi::Surface> const& surface){
-            EXPECT_THAT(surface.get(), Eq(stub_surface1.get()));
-        });
+        EXPECT_THAT(stack.input_surface_at({}).get(), Eq(stub_surface1.get()));
```

Comment:
Although it's a bit more effort, I feel we should be actually passing a non-default `Point` here (and in the other tests) rather than relying on `{}` to somehow be inside the `{}` size, `{}` position stub surface.

As it is, it looks to me like these tests erroneously pass? All the surfaces are 0-sized, and `Rectangle::contains` *should* return false for zero-sized rectangles?